apis/metal3.io/v1alpha1/hostupdatepolicy_types.go

```diff
@@ -0,0 +1,91 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HostUpdatePolicy enumerates the allowed host update policies
+type UpdatePolicy string
+
+const (
+	HostUpdatePolicyOnPreparing    UpdatePolicy = "onPreparing"
+	HostUpdatePolicyOnReboot       UpdatePolicy = "onReboot"	
+)
+
+// NOTE(janders) based on https://github.com/metal3-io/metal3-docs/pull/411/
+
+// HostUpdatePolicySpec defines the desired state of HostUpdatePolicy
+type HostUpdatePolicySpec struct {
+	// defines policy for changing firmware settings
+	// +optional
+	// +kubebuilder:validation:Enum="onPreparing";"onReboot"
+	FirmwareSettings UpdatePolicy `json:"firmwareSettings,omitempty"`
+
+	// defines policy for updating firmware
+	// +optional
+	// +kubebuilder:validation:Enum="onPreparing";"onReboot"
+	FirmwareUpdates UpdatePolicy `json:"firmwareUpdates,omitempty"`
+}
+
+// HostUpdatePolicyStatus defines the observed state of HostUpdatePolicy.
+type HostUpdatePolicyStatus struct {
+	// NOTE (janders) is this what we need to have empty status?
+	// if so I will remove the entire section
```

Comment:
Yes, I think you can leave the empty structure.

Reply:
Thank you, removing the commented out code.